### PR TITLE
Fix: prevent accessing undefined data-file-id attribute

### PR DESCRIPTION
**Pull request by olivier-pierre**
When using featherlight by myself (Adding `$('.selector').featherlightGallery()` in my theme, I got the following error: 
```
Uncaught TypeError: Cannot read property 'id-undefined' of undefined
```

### CODE REVIEW

File: js/featherlight.js [hunk 1]
```diff
@@ -21,25 +21,27 @@ Backdrop.behaviors.featherlight = {
...
       var caption = false;
       var img = this.$currentTarget.find('img');
-      var key = 'id-' + img.attr('data-file-id');
-      var captionAttr = settings.featherlight.files[key];
-
-      if (typeof captionAttr !== 'undefined') {
-        // Caption attribute taken from field formatter settings.
-        if (captionAttr) {
-          var caption = img.attr(captionAttr);
+      if (img[0].hasAttribute('data-file-id')) {
```
Comment (ghost) [inline js/featherlight.js:24]: In what situation would `img` be an array...? I can't replicate this.

File: js/featherlight.js [hunk 2]
```diff
@@ -21,25 +21,27 @@ Backdrop.behaviors.featherlight = {
...
+        var key = 'id-' + img.attr('data-file-id');
+        var captionAttr = settings.featherlight.files[key];
```
Comment (ghost) [inline js/featherlight.js:26]: The only error I can produce is: `TypeError: settings.featherlight.files is undefined`.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@olivier-pierre Can you please test the following re-factored code and let me know if that works for you instead? If so, I'll make a new PR and commit it.

``` js
$.featherlight.prototype.afterContent = function() {
  var captionAttr = false;
  var caption = false;
  var img = this.$currentTarget.find('img');

  // Get caption attribute.
  if (settings.featherlight.hasOwnProperty('files')) {
    if (typeof img.attr('data-file-id') !== 'undefined') {
      var key = 'id-' + img.attr('data-file-id');
      captionAttr = settings.featherlight.files[key];
    }
  }
  else {
    if (settings.featherlight.caption_attribute) {
      captionAttr = settings.featherlight.caption_attribute;
    }
  }

  // Get and set caption.
  if (captionAttr) {
    caption = img.attr(captionAttr);

    if (caption) {
      this.$instance.find('.featherlight-caption').remove();
      $('<div class="featherlight-caption">').text(caption).appendTo(this.$instance.find('.featherlight-content'));
    }
  }
};
```

**Comment by olivier-pierre** [comment]
No more error with your refactored code, thanks !

Note that `img` is always an array/object. It's a jQuery element, and `img[0]` or `img.get(0)` is the html element, if I understand it correctly.

But it doesn't really matter, your refactored code works perfectly, at least in my case !

**Comment by ghost** [comment]
I've made a separate branch with the new code: https://github.com/backdrop-contrib/featherlight/tree/fix_undefined_errors

I'll merge that soon if all goes well.

**Pull request closed without merging**